Sources/Shared/UI/Social/MessageComposerViewController.m

```diff
@@ -349,8 +350,14 @@ - (IBAction)onBtnSend:(id)sender
         if(self.attPhotoView.image)
         {
             FilesProxy *fileProxy = [FilesProxy sharedInstance];
+            NSString * photosFolderPath;
+            if (selectedSpace && selectedSpace.spaceId) {
+               photosFolderPath = [NSString stringWithFormat:@"%@/rest/private/jcr/%@/%@/Groups%@/Documents",[ApplicationPreferencesManager sharedInstance].selectedAccount.serverUrl,[ApplicationPreferencesManager sharedInstance].currentRepository, [ApplicationPreferencesManager sharedInstance].defaultWorkspace, selectedSpace.groupId];
+            } else {
+                photosFolderPath =  [NSString stringWithFormat:@"%@/Public", fileProxy._strUserRepository];
+            }
             
-            BOOL storageFolder = [fileProxy createNewFolderWithURL:[NSString stringWithFormat:@"%@/Public", fileProxy._strUserRepository] folderName:@"Mobile"];
+            BOOL storageFolder = [fileProxy createNewFolderWithURL:photosFolderPath folderName:@"Mobile"];
```

Comment:
The folder name should be defined as a constant in defines.h, e.g.
`#define MOBILE_UPLOAD_DEST_FOLDER @"Mobile"`